src/App.tsx

```diff
@@ -1,61 +1,132 @@
+import React, { FormEvent, useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+import { User } from './components/types/User';
+import { TodoList } from './components/TodoList/TodoList';
+import { Todo } from './components/types/Todo';
+
+const getUserById = (userId: number): User | null => {
+  const foundedUser = usersFromServer.find(user => user.id === userId);
+
+  return foundedUser || null;
+};
+
+const newTodoId = (todosArr: Todo[]): number => {
+  const todoIdArray = todosArr.map(todo => todo.id);
+  const newId = Math.max(...todoIdArray) + 1;
+
+  return newId;
+};
+
+const newTodos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
 
 export const App = () => {
+  const [todoTitle, setTodoTitle] = useState('');
+  const [selectedUserId, setSelectedUserId] = useState(0);
+  const [hasTitleError, setHasTitleError] = useState(false);
+  const [hasUserError, setHasUserError] = useState(false);
+  const [todos, setTodos] = useState<Todo[]>(newTodos);
+
+  const clearForm = () => {
+    setTodoTitle('');
+    setSelectedUserId(0);
+  };
+
+  const handleSubmit = (event: FormEvent) => {
+    event.preventDefault();
+
+    if (!todoTitle) {
```

Comment:
```suggestion
    if (!todoTitle.trim()) {
```
<img width="121" alt="image" src="https://user-images.githubusercontent.com/55802630/185101095-561af5c4-6455-4190-b06d-7f910415cd49.png">